File: ShoppingCart.cpp
#include "ShoppingCart.h"
#include <iostream>

void ShoppingCart::addItem(Item* item) {
    items.push_back(item);
}

void ShoppingCart::displayItems(std::ostream& os) const {
    for (auto item : items) {
        item->display(os);
    }
}

void ShoppingCart::accept(Visitor& visitor) {
    for (auto item : items) {
        item->accept(visitor);
    }
}

File: ShoppingCart.java
import java.util.*;

public class ShoppingCart {
    private List<Item> items = new ArrayList<>();

    public void addItem(Item item) {
        items.add(item);
    }

    public void displayItems() {
        items.forEach(Item::display);
    }

    public void accept(Visitor visitor) {
        items.forEach(item -> item.accept(visitor));
    }
}
